sources/packages/web/src/components/aest/students/bypassedRestrictions/HistoryBypassedRestrictions.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <v-card>
+    <v-container>
+      <body-header title="History of Bypassed Restrictions" class="m-1">
+        <template #actions>
+          <check-permission-role :role="Role.StudentAddRestriction">
+            <template #="{ notAllowed }">
+              <v-btn
+                @click="addBypassedRestriction"
+                class="float-right"
+                color="primary"
+                data-cy="addRestrictionButton"
+                prepend-icon="fa:fa fa-plus-circle"
+                :disabled="notAllowed"
+                >Bypass A Restriction</v-btn
+              >
+            </template>
+          </check-permission-role>
+        </template>
+      </body-header>
+      <content-group class="mt-4">
+        <toggle-content
+          :toggled="!bypassedRestrictions.length"
+          message="No bypassed restrictions found."
+        >
+          <DataTable
+            :value="bypassedRestrictions"
+            :paginator="true"
+            :rows="DEFAULT_PAGE_LIMIT"
+            :rowsPerPageOptions="PAGINATION_LIST"
+            :totalRecords="bypassedRestrictions.length"
+          >
+            <template #empty>
+              <p class="text-center font-weight-bold">No records found.</p>
+            </template>
+            <Column
+              field="restrictionType"
+              header="Restriction Type"
+              :sortable="true"
+            ></Column>
+            <Column field="restrictionCode" header="Restriction Code"></Column>
+            <Column header="View Details">
+              <template #body="{ data }">
+                <v-btn
+                  @click="$emit('viewBypassDetails', data.id)"
```

Comment:
I am not sure why we have to emit to parent component from here. 

Even if the line 45 doesn't exist without `@click` it should be fine.

Reply:
@dheepak-aot I didn't get it. `@click` would be needed for the `View Details` which might not be a part of this PR but it would be needed in future, right?

Reply:
Removed. @sh16011993 You're right. `@click` will be need in the following story to display a model for details.